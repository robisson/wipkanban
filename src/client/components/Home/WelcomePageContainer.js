import WelcomePage from "./WelcomePage";
import { connect } from "react-redux";
import { logout } from "../../actions/Login";

const mapStateToProps = ({ userReducer: { user } }) => ({ user });

const mapDispatchToProps = dispatch => {
  return {
    onLogout() {
      dispatch(logout());
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WelcomePage);
