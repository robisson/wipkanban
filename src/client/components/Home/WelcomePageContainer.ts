import WelcomePage from "./WelcomePage";
import { connect } from "react-redux";
import { logout } from "../../actions/Login";
import { updateAccountUser } from "../../actions/User";
import { Dispatch } from "redux";

const mapStateToProps = ({ userReducer: { user } }: any) => ({ user });

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onLogout() {
      dispatch(logout());
    },
    onUpdateAccountUser(user: Object) {
      dispatch(updateAccountUser(user));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WelcomePage);
