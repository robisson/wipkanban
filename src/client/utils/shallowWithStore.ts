import { shallow } from "enzyme";

export default (component:any, store:any) => {
  const context = {
    store
  };
  return shallow(component, { context });
};
