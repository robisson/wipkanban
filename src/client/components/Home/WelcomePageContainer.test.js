import React from "react";
import toJson from "enzyme-to-json";
import "../../setupTest";
import WelcomePageContainer from "./WelcomePageContainer";
import shallowWithStore from "../../utils/shallowWithStore";
import { createMockStore } from "redux-test-utils";

describe("<WelcomePageContainer /> Component", () => {
  let wrapper;

  it("should to render correctly", () => {
    const testState = {
      userReducer: {
        user: { firstAccess: true }
      }
    };

    const store = createMockStore(testState);
    wrapper = shallowWithStore(<WelcomePageContainer />, store);
    expect(wrapper.props().user).toEqual(undefined);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
