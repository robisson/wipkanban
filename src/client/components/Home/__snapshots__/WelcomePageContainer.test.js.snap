// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePageContainer /> Component should to render correctly 1`] = `
<WithStyles(WelcomePage)
  onLogout={[Function]}
  onUpdateAccountUser={[Function]}
  user={
    Object {
      "firstAccess": true,
    }
  }
/>
`;
